pkg/frontend/v2/frontend.go

```diff
@@ -176,6 +176,8 @@ func (f *Frontend) running(ctx context.Context) error {
 }
 
 func (f *Frontend) stopping(_ error) error {
+	// Wait until there are no more in-flight requests
+	f.requests.Wait()
```

Comment:
I would improve the comment to explain that we don't expect new queries to be added to `requests` after `Frontend.Stopping()` is called, because `Frontend.RoundTripGRPC()` checks whether the service is in the running state before taking the next request.

Reply:
Thanks @pracucci, will look into it.

Reply:
Done.